commit e752869b16555df7fd2c7aa69c7f38d8eebabfe9 (refs/changes/85/4485/1)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-07-05T13:33:49-04:00 (1 year, 3 months ago)
    
    TRITON-571 node-vmadm events should return the event stream object

diff --git a/README.md b/README.md
index d547011..b162f51 100644
--- a/README.md
+++ b/README.md
@@ -267,6 +267,39 @@ Callback:
     events` with the `-r` argument.
   - `obj.stop` a function to call to stop the event stream.
 
+This returns the underlying stream object that can be used to listen for
+errors.
+
+Example
+
+``` js
+var opts = {
+  log: log,
+  name: 'My Test Event Listener'
+};
+
+var vs = vmadm.events(opts, handler, ready);
+
+function ready(err, obj) {
+  // err is set if something failed during setup
+  // obj => Object
+  // obj.ev => Initial event with all VM information on the system
+  // obj.stop = Function to call to stop the event stream
+}
+
+function handler(ev) {
+  // Called whenever a new event is seen for any zone
+  // ev => Object
+  // ev.type => "create", "modify", "delete"
+  // ev.vm => VM payload or null (empty when ev.type == "delete")
+  // ev.zonename => zonename
+}
+
+vs.on('error', function (err) {
+  // called when an error is seen *after* the initial setup is complete
+});
+```
+
 ## Development
 
 Describe steps necessary for development here.
diff --git a/lib/index.sunos.js b/lib/index.sunos.js
index 5c7ab5b..d97e8b5 100644
--- a/lib/index.sunos.js
+++ b/lib/index.sunos.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -1179,6 +1179,8 @@ vmadm.stop = function vmStop(opts, callback) {
  *      - err {Error} set if an error occured that means the stream cannot be
  *      created
  *      - stop {Function} function to stop the event stream
+ * @return vs {VmadmCLIEventStream}
+ *      - Can be used to listen for errors, ex: `vs.on('error', ...)`
  */
 vmadm.events = function vmEvents(opts, handler, callback) {
     var vs;
@@ -1217,6 +1219,8 @@ vmadm.events = function vmEvents(opts, handler, callback) {
         log.debug('vmadm events stop called');
         vs.stop();
     }
+
+    return vs;
 };
 
 function VmadmCLIEventStream(opts) {
diff --git a/package.json b/package.json
index 1f4551c..03bce88 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "vmadm",
-  "version": "1.0.2",
+  "version": "1.1.0",
   "description": "Node.js wrapper around SmartOS vmadm(1M)",
   "main": "lib/index.js",
   "scripts": {
